src/coreclr/tools/ILVerification/AccessVerificationHelpers.cs

```diff
@@ -22,6 +22,14 @@ internal static bool CanAccess(this TypeDesc currentClass, TypeDesc targetClass)
             if (targetClass.IsParameterizedType)
                 return currentClass.CanAccess(((ParameterizedType)targetClass).ParameterType);
 
+            if (targetClass.IsFunctionPointer)
+            {
+                var signatureBeingPointed = ((FunctionPointerType)targetClass).Signature;
+
+                return currentClass.CanAccess(signatureBeingPointed.ReturnType) &&
```

Comment:
Does this need to de-reference `IsByRef` types, like what is done for the regular arguments?

Reply:
I was assuming that `CanAccess()` will cover all relevant cases. Is it special case for function pointer's ReturnType or should we check it in CanAccess? (here we are making a recursive call for checking any TypeDesc)

Reply:
I would expect the logic for both methods signatures and function pointer signatures to be exactly same.

I think it would be best to have a single `bool CanAccessSignature(this TypeDesc currentType, MethodSignature targetMethod)` method that will check both the return value and the arguments; and that will be used for both the methods and function pointers.

Reply:
Wrapped return type checks in `CanAccessSignature`.